Running-Mastodon/Production-guide.md

```diff
@@ -1,20 +1,190 @@
+**Disclaimer:**
 
+This guide was written for [Ubuntu Server 16.04](https://www.ubuntu.com/server), you may run into issues if you are using another operating system. We are welcoming contributions for guides to other distributions.
+
+This document is also written with the expectation that you have a technical level high enough to administrate Linux servers.
+
+## What is this guide?
+
+This guide is a walk through of the setup process of a [Mastodon](https://github.com/tootsuite/mastodon/) instance.
+
+We use example.com to represent a domain or sub-domain. Example.com should be replaced with your instance domain or sub-domain.
+
+## Prerequisites
+
+You will need the following for this guide:
+
+- A server running [Ubuntu Server 16.04](https://www.ubuntu.com/server).
+- Root access to the server.
+- A domain or sub-domain to use for the instance.
+
+## DNS
+
+DNS records should be added before anything is done on the server.
+
+The records added are:
+
+-  A record (IPv4 address) for example.com
+-  AAAA record (IPv6 address) for example.com
+
+> ### A Helpful And Optional Note
+>
+> Using `tmux` when following through with this guide will be helpful.
+> 
+>
+> Not only will this help you not lose your place if you are disconnected, it will let you have multiple terminal windows open for switching contexts (root user versus the mastodon user).
+>
+> You can install [tmux](https://github.com/tmux/tmux/wiki) from the package manager:
+>
+> ```sh
+> apt -y install tmux
+> ```
+
+## Dependency Installation
+
+All dependencies should be installed as root.
+
+### node.js Repository
+
+You will need to add an external repository so we can have the version of [node.js](https://nodejs.org/en/) required.
+
+Download this script:
+
+```sh
+wget https://deb.nodesource.com/setup_6.x
+```
+
+Review the script download before running:
+
+```sh
+less setup_6.x
+```
+
+Once you have reviewed the script, run it
+
+```sh
+bash setup_6.x
+```
+
+The [node.js](https://nodejs.org/en/) repository is now added.
+
+###  Yarn Repository
+
+Another repository needs to be added so we can get the version of [Yarn](https://yarnpkg.com/en/) used by [Mastodon](https://github.com/tootsuite/mastodon/).
+
+This is how you add the repository:
+
+```sh
+apt -y install curl
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+apt update
+```
+
+### Various Other Dependancies
+
+Now you need to install [Yarn](https://yarnpkg.com/en/) plus some more software.
+
+```sh
+apt -y install imagemagick ffmpeg libpq-dev libxml2-dev libxslt1-dev file git g++ libprotobuf-dev protobuf-compiler pkg-config nodejs gcc-6 autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 libgdbm-dev nginx redis-server redis-tools postgresql postgresql-contrib nginx letsencrypt yarn
```

Comment:
Yes, I opted to add those in as I felt it would be better to install all dependencies that we can install with apt in one go.

I concur with @nightpool that we need an explanation for all the software in that list. Let me write something to that order.